# The Gene Pool Model

## Background

Evolutionary biology helps us understand how life's genetic makeup changes over time. One important idea is the "gene pool," which contains all the different versions of genes within a group of organisms. These genes determine traits and characteristics. To grasp how genes evolve, we need to study how their frequencies change in a population.

In this practical, we'll use the R programming language to simulate how genes change in a population over generations. By changing some factors, you'll see how genetic diversity shifts in different situations. This exercise will deepen your understanding of evolution and give you practical skills to explore real genetic questions. So, let's start this journey into the world of genes and evolution, using code and data to learn how it all works.


## A simple model

### The gene pool

To establish the gene pool we first set the initial population size. I set this to be reasonably large, at 500 individuals.

```{r}
pop_size <- 500
```

There are two alleles (gene variants) per individual: one allele came from the father, and one from the mother.

Therefore, the gene pool contains 2 x N alleles (in this case, `r pop_size * 2` alleles).

```{r}
pool_size <- pop_size * 2
```

I now set the initial allele frequency to be 0.5. You can use other values, but 0.5 is best to start with.

```{r}
initial_frequency_A <- 0.5
initial_frequency_a <- 1 - initial_frequency_A
```

We can now calculate the **number** of *A* and *a* alleles, by multiplying the frequency by the pool size and rounding the result to the closest whole number.

```{r}
number_A <- round(initial_frequency_A * pool_size)
number_a <- round(initial_frequency_a * pool_size)
```


Following this, the gene pool can be filled like this. 
What we are saying here is "repeat `A`, `number_A` times, then repeat `a` `number_a` times"

```{r}
gene_pool <- c(rep("A", number_A), rep("a", number_a))
table(gene_pool)
```


## Projecting allele frequency over one time step.

Lets assume the population size remains constant at N = `r pop_size`.

During the next time step (i.e. 1 generation), the individuals in the population mate with each other randomly. We can use the `sample` function to simulate this.

This code is saying "**sample, from the gene pool, `r 2*pop_size` alleles**". We need to sample $2 \times$ the population size because each individual contains two alleles.

```{r}
new_gene_pool <- sample(gene_pool, 2 * pop_size, replace = TRUE)
```

Let's look at the new gene pool. It should be similar, but probably not exactly the same as the initial gene pool.

```{r}
table(new_gene_pool)
```

There are a total of `r pop_size` individuals, but the numbers are (probably) not exactly the same as in the initial gene pool. The reason the gene pool is (probably) not identical, is that it is a random sample, not simply a carbon copy.

**Try to vary the population size and do this several times at small and large population sizes.** You should notice that the frequencies/numbers are more similar when you repeat them for large population sizes than for small population sizes.

## Simulation through time

To simulate the passage of time, we need to do this sampling procedure many times.
We can do this using a loop in R. 

In our loop, we will repeat this gene pool sampling process many times (one time per generation). Each time we go through the loop, we sample alleles from the *old* gene pool from time $n$ to create a *new* gene pool for time $n+1$. And each time through the loop we calculate the allele frequency for A so we can track how it changes through time.

Let's set up a simulation that runs for 50 generations.
You can cut-paste this code, which is a complete gene-pool model.

```{r}
# Set number of generations
n_gen <- 50
pop_size <- 100

# Set up an empty vector of length n_gen to contain results
allele_freq_A <- rep(NA, n_gen)

# Put in an initial value for frequency of A
allele_freq_A[1] <- 0.5

# Establish the initial gene pool and name it gene_pool_time_n
# This part looks complicated, but I am just condensing the code from above into 1 line.
gene_pool_time_n <- c(rep("A", round(2 * allele_freq_A[1] * pop_size)), 
                    rep("a", round(2 * (1-allele_freq_A[1]) * pop_size)))

for (i in 2:n_gen) {
  gene_pool_time_n_plus_1 <- sample(gene_pool_time_n, 
                                    pop_size * 2, 
                                    replace = TRUE)
  gene_pool_table <- table(gene_pool_time_n_plus_1)
  allele_freq_A[i] <- gene_pool_table[2] / sum(gene_pool_table)

  # Replace the old gene pool with the new one.
  gene_pool_time_n <- gene_pool_time_n_plus_1
}
```


Now we can plot this result.

```{r}
plot(1:n_gen, allele_freq_A, type = "l", xlab = "time", ylim = c(0, 1))
```

**Tip: make a new script and paste the code loop, and the plot command so that you can run both together easily**

## What next?

Well done for making it this far. R programming is not for the fainthearted.

Next up, I want you to take the code, and try modifying the parameters to explore what would happen in different situations.

### Situation 1: small population size

Repeat the simulation using the code above using small population size, and large population size.


 - Is stochasticity in genetic diversity higher or lower with a small population size?
 - Is the probability of losing genetic variation higher or lower with a small population size?
 - Is "fixation" more or less likely with a small population?
 

### Situation 2: a genetic bottle neck

For this, we start with a large population size, but then impose a bottle neck.
We can add a line of code into the loop part of the model to simulate that.

This code means, if the generation time is 5, make the population size 10.

`if(i == 5){pop_size = 10}`

- How does a genetic bottle neck impact the genetic make-up of the population?
- How might a bottle neck impact the probability, over the long term, of genetic fixation?

